### PR TITLE
Implement Feature Request to make Data return slice of defined DataStruct

### PR DESCRIPTION
**Pull request by propersam**
#1 
In this commit:
[X] Break code base into 2 files
[X] Add New Method and function to Process Data and return []interface (slice of defined data struct type)
[X] Add Unit test

Any Area of code you want me to explain..can you point it out as you do a code review...just comment on those lines as issues for resolve or something. and I will try to explain what it does. 

update: 
[X] Add Integration Test

### CODE REVIEW

**Comment by propersam** [comment]
Here is a main.go sample code you can use to test

```
package main

import (
	"fmt"

	"github.com/davecgh/go-spew/spew"
	lasgo "github.com/laslibs/las-go"
)

type dataRow struct {
	Dept          float64 `las:"DEPT"`
	Perm          string  `las:"Perm"`
	Gamma         float64 `las:"Gamma"`
	Porosity      string  `las:"Porosity"`
	Fluvialfacies float64 `las:"Fluvialfacies`
	NetGross      string  `las:"NetGross`
}

func main() {
	las, err := lasgo.Las("./sample/A10.las")
	if err != nil {
		panic(err)
	}

	hds, _ := las.Header()
	fmt.Println(hds)

	// opts := &lasgo.DataOptions{ConcreteStruct: dataRow{}, DecoderConfig: &lasgo.StructorConfig{
	// 	DecodeHook:       mapstructure.StringToTimeHookFunc(time.RFC3339),
	// 	WeaklyTypedInput: true}}

	opts := &lasgo.DataOptions{ConcreteStruct: dataRow{}}

	data := las.DataStruct(opts)

	// for _, d := range data {

	// 	switch d := d.(type) {
	// 	case *dataRow:
	// 		fmt.Println(*d)
	// 	}
	// }

	spew.Dump(data) // spew is like a data print logger
}

```

**Comment by propersam** [comment]
If you encounter error anywhere..lemme know here..so i can try to fix ASAP

**Comment by iykekings** [comment]
wow, that is so nice..... 🔥 🔥 
I love this.

**Comment by propersam** [comment]
I hope you know go.mod will not allow you test other branch than master

to run code from another branch you have to make sure your code is within src directory (like *go/src/github/laslibs/las-go*) and within this directory you checkout out to `DataStruct` branch

your main.go file can be in any other directory but also within *go/src/* directory

**Review by iambenkay — APPROVED** [review]
This is fantastic. You worked magic with the reflect package, well done!!

**Review by iambenkay — DISMISSED** [review]

File: las.go [hunk 1]
```diff
@@ -97,6 +58,41 @@ func (l *LasType) Data() [][]string {
...
+		o     *DataOptions
+		store [][]string
+	)
+
+	if opt != nil {
+		o = opt
+	}
+
+	store = chunk(sBs, len(hds))
+
```
Comment (iykekings) [inline las.go]: @propersam can we replace this block with `store := *l.Data()`

File: test/integration_test.go [hunk 2]
```diff
@@ -48,3 +51,163 @@ func TestWrap(t *testing.T) {
...
+
+	actual := las.DataStruct(opts)
+
+	spew.Dump(actual)
+
+	if !cmp.Equal(expected, actual) {
+		t.Errorf("wrong val: expected: %T %+v actual: %T %v\n", expected, expected, actual, actual)
+	}
+
+}
```
Comment (iykekings) [inline test/integration_test.go:213]: You did a great job with these tests 🔥

**Review by iykekings — CHANGES_REQUESTED** [review]

File: las.go (hunk 1 above)
Comment (propersam) [reply las.go]: hmm...that's possible ...i also see repitition... lemme do that asap

File: test/integration_test.go (hunk 2 above)
Comment (propersam) [reply test/integration_test.go]: @iykekings i cnt send msgs nw... here is d link(i'll delete soon)
https://whereby.com/suretrade

Comment (iykekings) [reply test/integration_test.go:213]: Oh I didn't know you sent it here

**Review by iykekings — APPROVED** [review]

**Pull request merged**